nautobot/dcim/filters/__init__.py

```diff
@@ -1116,26 +1122,39 @@ class Meta:
             "interface_redundancy_groups",
         ]
 
+    def generate_query_filter_device(self, value):
+        if not hasattr(value, "__iter__") or isinstance(value, str):
+            value = [value]
+
+        device_ids = set(str(item) for item in value if is_uuid(item))
+        device_names = set(str(item) for item in value if not is_uuid(item))
+        devices = Device.objects.filter(Q(name__in=device_names) | Q(pk__in=device_ids))
+        all_interface_ids = []
+        for device in devices:
+            all_interface_ids.extend(device.vc_interfaces.values_list("id", flat=True))
+        return Q(pk__in=all_interface_ids)
+
     def filter_device(self, queryset, name, value):
-        try:
-            devices = Device.objects.filter(**{f"{name}__in": value})
-            all_interface_ids = []
-            for device in devices:
-                all_interface_ids.extend(device.all_interfaces.values_list("id", flat=True))
-            return queryset.filter(pk__in=all_interface_ids)
-        except Device.DoesNotExist:
-            return queryset.none()
+        # Include interfaces belonging to virtual chassis members
```

Comment:
Comment seems a bit out of place here, should it be moved up to line 1132/1133?

Reply:
This is where it was at before but I'm ok moving it or removing it completely. I think the comment above the filter definitions is sufficient